course/src/main/java/in/testpress/course/ui/VideoDownloadQualityChooserDialog.kt

```diff
@@ -66,16 +66,20 @@ class VideoDownloadQualityChooserDialog(val content: DomainContent) : DialogFrag
 
     private fun setOnClickListeners() {
         okButton.setOnClickListener {
-            val downloadRequest = videoDownloadRequestCreateHandler.buildDownloadRequest(overrides)
-            onSubmitListener?.invoke(downloadRequest)
+            if (::overrides.isInitialized) {
```

Comment:
In your PR description, you only mentioned the progress bar. If this is also necessary, can this be given in another PR.